fun will_throw
{

8}var exc = void
try {[will_throw()... ]
}
catch const   e{fun will_throw
{try {[ will_throw()... ]
}
catch const e {
  exc =delete e.a
}
}[~will_throw() ]}assert! exc
